app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -566,4 +586,61 @@ public CustomProgressDialog generateProgressDialog(int taskId) {
     public boolean isBackEnabled() {
         return false;
     }
+    
+    private void refreshView() {
+        Spinner spinner = (Spinner) findViewById(R.id.app_selection_spinner);
+        TextView message = (TextView) findViewById(R.id.welcome_msg);
+        ArrayList<ApplicationRecord> readyApps = CommCareApplication._().getReadyAppRecords();
+        if (readyApps.size() <= 1) {
+            spinner.setVisibility(View.GONE);
+            message.setText(R.string.login_welcome_single);
+            return;
+        }
+        message.setText(R.string.login_welcome_multiple);
+        ArrayList<String> appNames = new ArrayList<>();
+        ArrayList<String> appIds = new ArrayList<>();
+        for (ApplicationRecord r : readyApps) {
+            String name = r.getDisplayName();
+            appNames.add(name);
+            appIds.add(r.getUniqueId());
+            namesToRecords.put(name, r);
+        }
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, R.layout.spinner_text_view, appNames);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(adapter);
+        spinner.setOnItemSelectedListener(this);
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);

```

Comment:
Hm so again, I think it would be good to not introduce more ways in which LoginActivity is dependent upon there being a currently-seated app. Since onItemSelected() (which seats the selected app) will always be called when the spinner's selection is set, there is no danger of inconsistency here. If the currently seated app is different from the last selected app (which is really only possible if someone did something very unnatural like left the login screen, went to the manager and ran an update for an app other than the last selected one, and then came directly back to the login screen), onResume() would call refreshView(), which would trigger onItemSelected(), bringing the spinner's selection and the currently-seated app back into sync.

Let me know if you still feel strongly about this though, could be persuaded otherwise.